crates/rust-analyzer/src/handlers.rs

```diff
@@ -1127,7 +1127,7 @@ pub(crate) fn publish_diagnostics(
         .map(|d| Diagnostic {
             range: to_proto::range(&line_index, d.range),
             severity: Some(to_proto::diagnostic_severity(d.severity)),
-            code: None,
+            code: d.code.map(|d| d.as_str().to_owned()).map(NumberOrString::String),
```

Comment:
In `map_rust_diagnostic_to_lsp()` this is simplified as: `code: code.clone().map(lsp_types::NumberOrString::String)`

Reply:
Oh right, I somehow missed that I can do that. Good point 👍

Reply:
Makes sense.